src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -243,11 +248,25 @@ public class PullRequestHandler : IHttpHandler
             success = RunGitCommands(new string[] { "branch --list" }, gitRepoPath, branchesString);
         }
 
-        executionResult.Append(branchesString);
+        executionResult.Append(branchesString.ToString());
 
         if (success)
         {
-            branches = branchesString.ToString().Split(new char[] { '\n', '\r', ' ', '*' }, StringSplitOptions.RemoveEmptyEntries);
+            tempBranches = branchesString.ToString().Split(new char[] { '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries);
+            List<string> branchesList = new List<string>();
+
+            // Parse for anything that implies a detached HEAD - those are not branches to clean up
+            for (int i = 0; i < tempBranches.Length; i++)
+            {
+                if (tempBranches[i].Contains("HEAD detached"))
+                {
+                    continue;
+                }
+
+                branchesList.Add(tempBranches[i].Trim('*', ' '));
```

Comment:
NIT: flipping the logic of `if` at line 261, `continue` can be replaced with this line... it should be still "clean looking"... not sure if you will consider linq clean looking though 😉 
`branches = tempBranches.Where(b => !b.Contains("HEAD detached")).Select(b => b.Trim('*', ' ')).ToArray();`